activerecord/test/cases/connection_adapters/merge_and_resolve_default_url_config_test.rb

```diff
@@ -10,13 +10,15 @@ def setup
         @previous_rack_env = ENV.delete("RACK_ENV")
         @previous_rails_env = ENV.delete("RAILS_ENV")
         @adapters_was = ActiveRecord::ConnectionAdapters.instance_variable_get(:@adapters).dup
+        @protocol_adapters = ActiveRecord.protocol_adapters.dup
```

Comment:
```suggestion
        @protocol_adapters = ActiveRecord.protocol_adapters.to_h
```

Should we cast this to a regular hash instead?

Also `InheritedOptions` use symbol keys, we probably want string keys here.

Reply:
`InheritedOptions` is indifferent access:

```
> ActiveRecord.protocol_adapters["mysql"]
=> "mysql2"
> ActiveRecord.protocol_adapters[:mysql]
=> "mysql2"
```

But I'm realizing both `.dup` and `.to_h` return empty hashes, which are both wrong, or at least unexpected. Honestly I'm having a hard time figuring out how to manage this class because `.keys` also returns an empty array.

Reply:
Oh I see now it's not indifferent access, but you're right it's symbol indexed with:
```
    def [](key)
      super(key.to_sym)
    end
```

Reply:
https://github.com/rails/rails/pull/50151